### DISCUSSION TITLE
How can we, as developers, understand the Nuxt codebase to contribute and elevate this framework to new heights?

### DISCUSSION THREAD

**Discussion by waleedtariq109** [nuxt/nuxt#25192] [General]
Hi everyone,

I want to know, and I'm sure many other developers, just like me, also want to contribute to the Nuxt ecosystem. The problem is, we don't know where to start – how to understand such a big codebase, what's the entry point, how Nuxt leverages Vue, and more. What's the best and most effective way to understand the Nuxt3 codebase so that we can also contribute to the Nuxt project?

**Comment by TheAlexLichter** [comment]
Thinking back to how I started - go in "small". Here are some suggestions:

* Try contributing to a [module](https://github.com/nuxt-modules/) first - that way it helps understanding where you can "hook into nuxt"
* Same goes for an [UnJS package](https://github.com/unjs/) as they build the foundation around Nuxt
* Another good option are [documentation issues](https://github.com/nuxt/nuxt/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Adocumentation+) - it would help a lot to add content to the Nuxt docs
* If you want to contribute code to the core repo, pick one ["good first issue"](https://github.com/nuxt/nuxt/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22) you want to tackle. These usually don't need that much deep knowledge
* Join the discord and ask for help in #contributing if you have any questions! The team and community is happy to help you
* If you want to dive deeper into the Nuxt codebase, take a look at [Daniel's streams](https://www.youtube.com/@danielroe/videos) where he regularly goes into fixing/improving Nuxt

**Reply by waleedtariq109** [reply]
Thanks @manniL

**Reply by renardsas** [reply]
And also these excellent videos 😁:
https://www.youtube.com/@TheAlexLichter
